### DISCUSSION TITLE
dry-run custom migration

### DISCUSSION THREAD

**Discussion by Tobbe** [prisma/prisma#6628] [Q&A] [answered]
I've written a migration where I copy some data from one table to another, and then dropping that first table. But my SQL is a but rusty, so I want to make sure the migration works before I go dropping my tables. Is there any way to dry-run migrations? Or any other way I can make sure it'll do what I want it to do before I risk losing any data?

**Comment by Tobbe** [comment]
I cloned my DB and ran the SQL statements manually against that DB. Not as seamless as one might wish, but it works...

**Comment by 2color** [comment, marked as answer]
Hey @Tobbe 👋,
Looks like you figured it out. But I'll share my perspective on this

Technically, running `prisma migrate dev` in development tests your migrations by first running them against a [shadow database](https://www.prisma.io/docs/concepts/components/prisma-migrate/shadow-database) to make sure they're valid. However, this won't necessarily help with data migrations.

My suggestion would be to dump the database, import it into a new one (essentially cloning it for testing purposes), and then use Prisma Migrate to run the migration against it. 

This is a slight improvement over running the SQL statement manually and closer to what it will actually look like once you carry out the migrations in production. 

I would suggest that you open a feature request explaining how you would see dry-running migrations that include data migrations in Prisma Migrate.

**Reply by Tobbe** [reply]
Thanks for your response. Don't know why I didn't think of switching the DATABASE_URL env var and running the migration against another DB to test. That's perfect! I'll do that next time, thank you :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
